apps/dapp-console-api/src/routes/faucet/FaucetRoute.ts

```diff
@@ -0,0 +1,466 @@
+        })
+        .refine((obj) => !ON_CHAIN_AUTH_MODES.includes(obj.authMode), {
+          message: 'Authentication mode is not a supported off-chain mode',
+        }),
+    )
+    .mutation<FaucetClaim>(async ({ ctx, input }) => {
+      const recipientAddress = input.recipientAddress as Address
+      const { chainId, authMode } = input
+
+      switch (authMode) {
+        case 'PRIVY':
+          break
+        default:
+          throw new TRPCError({
+            code: 'INTERNAL_SERVER_ERROR',
+            message: 'Authentication mode is not supported',
+          })
+      }
+
+      console.log('got here')
+
+      const faucet = this.faucets.find((f) => f.chainId === chainId)
+
+      if (!faucet) {
+        throw new TRPCError({
+          code: 'NOT_FOUND',
+          message: `No faucet found for chainId ${chainId}`,
+        })
+      }
+
+      let tx: Hex | undefined
+      let error: FaucetError | undefined
+      try {
+        tx = await faucet.triggerFaucetDrip({
+          userId: this.getUserIdForAuthMode(ctx.session, authMode),
+          recipientAddress,
+          authMode,
+        })
+      } catch (err) {
+        error = this.handleFaucetDripError(err, faucet.chainId)
+        return {
+          chainId: faucet.chainId,
+          error,
+          recipientAddress,
+          authMode,
+        }
+      }
+      return {
+        chainId: faucet.chainId,
+        tx,
+        etherscanUrl: tx && faucet.getBlockExplorerUrlTx(tx),
+        error,
+        amountDistributed: faucet.offChainDripAmount,
+        recipientAddress,
+        authMode,
+      }
+    })
+
+  public readonly handler = this.trpc.router({
+    [this.faucetsInfo]: this.faucetsInfoController,
+    [this.nextDripsRoute]: this.nextDripsController,
+    [this.onChainClaimsRoute]: this.onChainClaimsController,
+    [this.offChainClaimsRoute]: this.offChainClaimsController,
+  })
+
+  constructor(
+    trpc: Trpc,
+    private readonly faucets: Faucet[],
+  ) {
+    super(trpc)
+  }
+
+  /// The user id to store on-chain for tracking the last time a user that authed with
+  /// github received a faucet drip.
+  private readonly getUserIdForPrivyAuth = (userId: string) =>
+    keccak256(stringToHex(userId))
+
+  /// The user id to store on-chain for tracking the last time a user that authed with
+  /// the optimist nft received a faucet drip.
+  private readonly getUserIdForWalletAuth = (ownerAddress: Address) =>
+    keccak256(ownerAddress)
+
+  /// The user id to store on-chain for tracking the last time a user that authed with
+  /// worldId received a faucet drip.
+  private readonly getUserIdForWorldIdAuth = (nullifierHash: string) =>
+    keccak256(nullifierHash as Hex)
+
+  private readonly getUserIdForAuthMode = (
+    session: IronSession<SessionData>,
+    authMode: FaucetAuthMode,
+    walletAddress?: Address,
+  ) => {
+    switch (authMode) {
+      case 'PRIVY':
+        if (!session.user) {
```

Comment:
nit: this should also check that `privyDid` is defined:
```suggestion
        if (!session.user.privyDid) {
```